django_netjsonconfig/base/admin.py

```diff
@@ -200,6 +204,17 @@ def download_view(self, request, pk):
         return send_file(filename='{0}.tar.gz'.format(config.name),
                          contents=config_archive.getvalue())
 
+    def device_context_view(self, request, pk):
+        instance = get_object_or_404(self.model, pk=pk)
+        if hasattr(instance, 'config'):
+            context = instance.config.get_context()
+            status = 200
+        else:
+            context = {'error': 'device has no config'}
+            status = 403
```

Comment:
I'm not sure this would work, because we want to use this API also for new devices which will not have any config object set.

Can you try instantiating (without saving) a `Config` model instance, associate it to the device and call the `get_context()` on that?